reanalysis/seqr_interactions.py

```diff
@@ -0,0 +1,379 @@
+"""
+api experiments with an auth-blocked seqr instance
+
+this is just a sketch - no fancy requests handling
+"""
+
+import json
+from collections import defaultdict
+
+import click
+import requests
+
+import google.auth.exceptions
+import google.auth.transport.requests
+from google.oauth2 import service_account
+
+from cpg_utils.config import get_config
+from sample_metadata.apis import ProjectApi
+
+from reanalysis.html_builder import make_coord_string
+
+
+# grab this from config in future
+# SEQR_INSTANCE = 'https://seqr-reanalysis-dev.populationgenomics.org.au'
+SEQR_INSTANCE = 'http://localhost:3000'
+
+SEQR_AUDIENCE = (
+    '1021400127367-4vch8s8kc9opeg4v14b2n70se55jpao4.apps.googleusercontent.com'
+)
+CREATE_ENDPOINT = '/api/saved_variant/create'
+SAMPLES_ENDPOINT = '/api/project/{projectGuid}/samples/sa'
+VARIANTS_ENDPOINT = '/api/project/{projectGuid}/saved_variants/'
+AIP_VARIANTS_ENDPOINT = '/api/project/{projectGuid}/aip_saved_variants'
+
+# mapping of AIP categories to seqr tag names
+AIP_CATEGORY_FLAGS = {
+    'category_1': 'AIP Cat 1',
+    'category_2': 'AIP Cat 2',
+    'category_3': 'AIP Cat 3',
+    'category_5': 'AIP Cat 5',
+    'category_support': 'AIP Cat Support',
+}
+AIP_DE_NOVO_FLAG = {'category_4': 'AIP Cat 4'}
+
+
+def get_token(
+    cred_file: str = '/Users/mwelland/Downloads/seqr-308602-713cfeb8eeb6.json',
+):
+    """
+    borrowed/stolen from github.com/populationgenomics/sample-metadata/blob/
+    81e7dddead12729e78280eaf0911a2cd59e34c1c/scripts/sync_seqr.py#L549
+    """
+
+    with open(cred_file, encoding='utf-8') as f:
+
+        info = json.load(f)
+        credentials_content = (info.get('type') == 'service_account') and info or None
+        credentials = service_account.IDTokenCredentials.from_service_account_info(
+            credentials_content, target_audience=SEQR_AUDIENCE
+        )
+        auth_req = google.auth.transport.requests.Request()
+        credentials.refresh(auth_req)
+        return credentials.token
+
+
+def get_project_from_metamist() -> str | None:
+    """
+    get the seqr project ID for this project/seq type
+    Returns:
+        str: the Seqr project GUID or None
+    """
+    project_dataset = get_config()['workflow']['dataset']
+    proj_seqr_key = f'seqr-project-{get_config()["workflow"]["sequencing_type"]}'
+
+    # filter the returned projects
+    projects = [
+        proj
+        for proj in ProjectApi().get_seqr_projects()
+        if proj.get('name') == project_dataset
+    ]
+
+    # WALRUS
+    if results := len(projects) != 1:
+        raise Exception(f'Expected one project for {project_dataset}, found {results}')
+
+    project = projects[0]
+
+    if proj_seqr_key in project['meta'] and project['meta'].get('is_seqr'):
+        return project['meta'].get(proj_seqr_key)
+
+    return None
```

Comment:
You're already getting seqr projects above:
```suggestion
    return project['meta'].get(proj_seqr_key)
```

Reply:
Heh, I was wondering if there was any possibility of having a populated project and `is_seqr=False`, but that wouldn't make any sense...